app/uk/gov/hmrc/economiccrimelevyregistration/services/RegistrationValidationService.scala

```diff
@@ -16,79 +16,131 @@
 
 package uk.gov.hmrc.economiccrimelevyregistration.services
 
-import cats.data.ValidatedNec
+import cats.data.Validated.Valid
+import cats.data.ValidatedNel
 import cats.implicits._
-import uk.gov.hmrc.economiccrimelevyregistration.models.AmlSupervisorType.{FinancialConductAuthority, GamblingCommission}
+import uk.gov.hmrc.economiccrimelevyregistration.models.AmlSupervisorType.{FinancialConductAuthority, GamblingCommission, Hmrc}
 import uk.gov.hmrc.economiccrimelevyregistration.models.EntityType._
 import uk.gov.hmrc.economiccrimelevyregistration.models.errors.DataValidationError
+import uk.gov.hmrc.economiccrimelevyregistration.models.errors.DataValidationError._
 import uk.gov.hmrc.economiccrimelevyregistration.models.grs.{IncorporatedEntityJourneyData, PartnershipEntityJourneyData, SoleTraderEntityJourneyData}
-import uk.gov.hmrc.economiccrimelevyregistration.models.{AmlSupervisor, Registration}
+import uk.gov.hmrc.economiccrimelevyregistration.models.integrationframework.LegalEntityDetails.CustomerType
+import uk.gov.hmrc.economiccrimelevyregistration.models.integrationframework._
+import uk.gov.hmrc.economiccrimelevyregistration.models.{AmlSupervisor, ContactDetails, EclAddress, Registration}
+import uk.gov.hmrc.economiccrimelevyregistration.utils.{SchemaLoader, SchemaValidator}
 
+import java.time.format.DateTimeFormatter
+import java.time.{Clock, Instant, LocalDate, ZoneId}
 import javax.inject.Inject
 
-class RegistrationValidationService @Inject() () {
+class RegistrationValidationService @Inject() (clock: Clock, schemaValidator: SchemaValidator) {
 
-  type ValidationResult[A] = ValidatedNec[DataValidationError, A]
+  type ValidationResult[A] = ValidatedNel[DataValidationError, A]
 
-  def validateRegistration(registration: Registration): ValidationResult[String] =
+  def validateRegistration(registration: Registration): ValidationResult[EclSubscription] =
+    transformToEclSubscription(registration) match {
+      case Valid(eclSubscription) =>
+        schemaValidator.validateAgainstJsonSchema(
+          eclSubscription,
+          SchemaLoader.loadSchema("create-ecl-subscription-request.json")
```

Comment:
Is there any benefit in placing the schema filename in a val?

Reply:
It's only used here, so unless you can think of a good reason I'll probably leave it?